Re-enable skipped tests and improve type coverage

gsteel opened the pull request:
|    Q          |   A
|-------------- | ------
| BC Break      | maybe
| QA            | yes

### Description

Mostly just QA re-enabling tests and cleaning up modified tests for CS and psalm.

Potential BC break is changing ExceptionInterface to extend from Throwable - if this _is_ a BC break, I can revert and update baseline.

gsteel committed:
Re-enable and clean up test for head title translation. Adds extensions required in dev to CI.

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Replace prophecy with PHPUnit mock

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Remove unnecessary white space, comments and tearDown()

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
All tests pass without being skipped

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Tear down is unnecessary. Resetting the default partial is unnecessary.

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Simplify exception related tests. Remove unnecessary @return

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
These tests pass, even with prefer-lowest.

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Refactors response header assertions to a private assertion and cleans up other psalm related errors

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Potential BC Break: Changes exception interface to extend Throwable

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Updates psalm-baseline.xml

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel commented:
Tests are failing on 8.1 because Paginator needs to be compatible with 8.1

froschdesign commented:
@gsteel 
> Tests are failing on 8.1 because Paginator needs to be compatible with 8.1

There are already some pollyfills:

https://github.com/laminas/laminas-view/tree/2.14.x/polyfill

Maybe the problem can be solved this way _for the moment_.

(I will create issue reports in the related component issue trackers to remove the pollyfills with an update.)

froschdesign commented:
@gsteel 
I found another problem:

https://github.com/laminas/laminas-view/blob/7ba573a745c16b4359fac558b4ef8af7426943a8/test/Helper/LayoutTest.php#L17-L34

If you could take a look at it, I would be very grateful.

gsteel committed:
Removes void return doc blocks in tests replacing with void return type where relevant

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Removes unnecessary tearDowns in tests

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Adds uninitialised properties and reduces psalm errors

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Adds testing shims for Paginator and ArrayAdapter which are not currently compatible with 8.1

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel committed:
Chop down long lines

Signed-off-by: George Steel <george@net-glue.co.uk>

gsteel commented:
> @gsteel
> 
> > Tests are failing on 8.1 because Paginator needs to be compatible with 8.1
> 
> There are already some pollyfills:
> 
> https://github.com/laminas/laminas-view/tree/2.14.x/polyfill
> 
> Maybe the problem can be solved this way _for the moment_.
> 
> (I will create issue reports in the related component issue trackers to remove the pollyfills with an update.)

Hi @froschdesign - I've added these polyfills and addressed the uninitialised props in Layout test

froschdesign commented on src/Exception/ExceptionInterface.php:
```diff
@@ -2,6 +2,8 @@
 namespace Laminas\View\Exception;
 
-interface ExceptionInterface
+use Throwable;
+
+interface ExceptionInterface extends Throwable
```
@Ocramius @boesing 
> Potential BC break is changing ExceptionInterface to extend from Throwable - if this is a BC break, I can revert and update baseline.

Any comments on this change?

Ocramius replied:
Potentially a BC break, but then `ExceptionInterface` is not supposed to be relied upon when implementing things downstream, but rather as a marker interface.

IMO acceptable, and I'm prepared to tell downstream consumers why they were wrong upfront, should there be problems :-)

froschdesign commented:
@gsteel 
> I've added these polyfills and addressed the uninitialised props in Layout test

I have already seen that. Great and thank you! 👍 

(I have the issue reports for the components on my list.)

Ocramius commented on polyfill/Paginator/ArrayAdapter.php:
```diff
@@ -0,0 +1,65 @@
+ * This class replaces the laminas-paginator equivalent during development only, for
+ * purposes of testing against PHP 8.1.
+ *
+ * @todo Remove when laminas-paginator has a release targeting PHP 8.1.
```
Can we make this `@internal` here?

Ocramius commented on polyfill/Paginator/ArrayAdapter.php:
```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Laminas\Paginator\Adapter;
```
We should `export-ignore` the `polyfill/` directory

froschdesign replied:
@Ocramius 
But this means that in other environments laminas-view does not work with the paginator and PHP 8.1.
I am a bit surprised because it was not excluded before.

Ocramius replied:
@froschdesign aren't the `polyfill/` classes just for test use? They were only loaded in `autoload-dev`

froschdesign replied:
You're absolutely right, of course!

gsteel committed:
Export ignore the polyfill directory. Remove unnecessary autoloader for tests. Mark all shim classes as internal

Signed-off-by: George Steel <george@net-glue.co.uk>

Ocramius reviewed (approved):
LGTM :+1:

Ocramius commented:
Oooof, there's no `2.15.x` :| 

This sort of stuff should really be in a minor, but `laminas/automatic-releases` (correctly) didn't create `2.15.x`, because `3.0.x` already exists.

Ocramius commented:
Thanks @gsteel!
